File: eminus5/Estudiantes/lib/domain/entities/actividad.dart
import 'entrega.dart';

class Actividad {
  final String id;
  final String titulo;
  final String descripcion;
  final DateTime fechaInicio;
  final DateTime fechaTermino;
  final double valorMaximo;
  List<Entrega> entregas; 

  Actividad({
    required this.id,
    required this.titulo,
    required this.descripcion,
    required this.fechaInicio,
    required this.fechaTermino,
    this.valorMaximo = 100, 
    List<Entrega>? entregas,
  }) : entregas = entregas ?? [];

  factory Actividad.fromJson(Map<String, dynamic> json) {
    return Actividad(
      id: json['id'] ?? '',
      titulo: json['titulo'] ?? '',
      descripcion: json['descripcion'] ?? '',
      fechaInicio: DateTime.parse(json['fechaInicio']),
      fechaTermino: DateTime.parse(json['fechaTermino']),
      valorMaximo: (json['valorMaximo'] ?? 100).toDouble(),
      entregas: json['entregas'] != null 
          ? (json['entregas'] as List).map((e) => Entrega.fromJson(e)).toList() 
          : [],
    );
  }
}

File: eminus5/campusvirtual/src/main/java/com/campusvirtual/backend/model/Actividad.java
package com.campusvirtual.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonIgnore;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "actividades")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Actividad {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String titulo;

    @Column(columnDefinition = "TEXT")
    private String descripcion;

    @Column(nullable = false)
    private LocalDateTime fechaInicio;

    @Column(nullable = false)
    private LocalDateTime fechaTermino;

    private double valorMaximo = 100.0;

    // Relación N:1 (Muchas actividades pertenecen a un Curso)


    // (En ambas clases, ve a la relación de "Curso" y añade @JsonIgnore)
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "curso_id", nullable = false)
    @JsonIgnore // Un mensaje o actividad no necesita imprimir todo el curso de vuelta
    private Curso curso;

    // Relación 1:N con Entregas
    @OneToMany(mappedBy = "actividad", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Entrega> entregas = new ArrayList<>();
}